lib/morpheme.js

```diff
@@ -125,17 +125,47 @@ export default class Morpheme {
    */
   matchingValues (feature) {
     let matches = []
+
     if (feature && this.features.hasOwnProperty(feature.type)) {
       const morphemeValue = this.features[feature.type]
-      for (const featureValue of feature.values) {
-        if (morphemeValue.values.includes(featureValue)) {
-          matches.push(featureValue)
-        }
+
+      let flagPush = false
+
+      if (feature.type !== 'full form' && morphemeValue.value === feature.value) {
+        flagPush = true
+      } else if (feature.type === 'full form' && (!this.extendedLangData || !this.extendedLangData.numeralFullFormValues) && morphemeValue.value.indexOf(feature.value) > -1) {
+        flagPush = true
+      } else if (feature.type === 'full form' && this.extendedLangData && this.extendedLangData.numeralFullFormValues && this.findFromNumeralFullFormValues(feature)) {
```

Comment:
@irina060981 and @kirlat I don't really understand why this extra step is necessary for the numerals. Why doesn't it work to check if the morphemeValue.values includes the form in the feature value? Isn't that essentially the same thing that is being checked in findFromNumeralFullFormValues ?

Reply:
I think I could describe:
Numerals (for example δύο (2)) have quite different forms for various gender/grmCase:

δύο (2)  - δύο (dual/accusative)
δύο (2)  - δυοῖν (dual/dative)
δύο (2)  - δυοῖν (dual/genitive)
δύο (2)  - δύο (dual/nominative)

So when we get form (on input) - δύο - it has for analysis all the array [ δύο, δυοῖν, μίαν, μιᾷ ....]
And it should get from matching these variants - δύο, δυοῖν
But these forms have no common parts - these forms quitely different.

That's why - I am passing to comparison - the pairs (hwd + form) in the additional object.
And in matching - I am checking - 

if it is full form based (as pronoun) - and then it uses simply checking morphemeValue.value = feature.value
Otherwise - I am using additional function - findFromNumeralFullFormValues - and it takes whole bunches - hdw - [form, form, ....] - and if both  morphemeValue.value and feature.value are from the same bunch - I consider them to be equal.

And this way I am getting such a distribution in inflection table.

If you think that it is not the best solution I will be happy to learn another approach.